packages/app/src/hooks/useFetchMetricResourceAttrs.tsx

```diff
@@ -0,0 +1,120 @@
+import { ResponseJSON } from '@clickhouse/client';
+import { chSql } from '@hyperdx/common-utils/dist/clickhouse';
+import { SourceKind } from '@hyperdx/common-utils/dist/types';
+import { TSource } from '@hyperdx/common-utils/dist/types';
+import { useQuery } from '@tanstack/react-query';
+
+import { getClickhouseClient } from '@/clickhouse';
+import { formatAttributeClause } from '@/utils';
+
+const METRIC_FETCH_LIMIT = 25;
+
+const extractAttributeKeys = (
+  attributesArr: {
+    ScopeAttributes?: object;
+    ResourceAttributes?: object;
+    Attributes?: object;
+  }[],
+  isSql: boolean,
+) => {
+  try {
+    const resultSet = new Set<string>();
+    for (const attribute of attributesArr) {
+      if (attribute.ScopeAttributes) {
+        Object.entries(attribute.ScopeAttributes).forEach(([key, value]) => {
+          const clause = formatAttributeClause(
+            'ScopeAttributes',
+            key,
+            value,
+            isSql,
+          );
+          resultSet.add(clause);
+        });
+      }
+
+      if (attribute.ResourceAttributes) {
+        Object.entries(attribute.ResourceAttributes).forEach(([key, value]) => {
+          const clause = formatAttributeClause(
+            'ResourceAttributes',
+            key,
+            value,
+            isSql,
+          );
+          resultSet.add(clause);
+        });
+      }
+
+      if (attribute.Attributes) {
+        Object.entries(attribute.Attributes).forEach(([key, value]) => {
+          const clause = formatAttributeClause('Attributes', key, value, isSql);
+          resultSet.add(clause);
+        });
+      }
+    }
+    return Array.from(resultSet);
+  } catch (e) {
+    console.error('Error parsing metric autocompleteattributes', e);
+    return [];
+  }
+};
+
+interface MetricResourceAttrsProps {
+  databaseName: string;
+  tableName: string;
+  metricType: string;
+  metricName: string;
+  tableSource: TSource | undefined;
+  isSql: boolean;
+}
+
+export const useFetchMetricResourceAttrs = ({
+  databaseName,
+  tableName,
+  metricType,
+  metricName,
+  tableSource,
+  isSql,
+}: MetricResourceAttrsProps) => {
+  const shouldFetch = Boolean(
+    databaseName &&
+      tableName &&
+      tableSource &&
+      tableSource?.kind === SourceKind.Metric,
+  );
+  return useQuery({
+    queryKey: ['metric-attributes', databaseName, tableName, metricType],
+    queryFn: async ({ signal }) => {
+      if (!shouldFetch) {
+        return [];
+      }
+
+      const clickhouseClient = getClickhouseClient();
+
+      const sql = chSql`
+        SELECT 
```

Comment:
do we want to do select distinct?

Reply:
updated in latest commit, as well as increasing the initial limit